crates/precompile/src/bls12_381/g1_msm.rs

```diff
@@ -67,8 +65,7 @@ pub(super) fn g1_msm(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         }
 
         g1_points.push(p0_aff);
-
-        scalars_bytes.extend_from_slice(&extract_scalar_input(encoded_scalar)?.b);
+        scalars.push(read_scalar(encoded_scalar)?);
```

Comment:
This now reads in a scalar and the rest of the API uses this scalar type. This means that you cannot use any of the methods which require scalars without calling read_scalar first, which checks that the byte slice is 32 bytes.

Reply:
Yeah this is a lot nicer!